src/test/kotlin/org/dashj/platform/dpp/Fixtures.kt

```diff
@@ -1,86 +1,332 @@
+    fun getIdentityFixtureTwo(): Identity {
         val json = File("src/test/resources/data/identity.json").readText()
         val jsonObject = JSONObject(json)
         val rawIdentity = jsonObject.toMap()
 
         return Identity(rawIdentity)
     }
 
+    fun getIdentityFixture(): Identity {
+
+        val id = Entropy.generateRandomIdentifier()
+
+        val rawIdentity = mapOf<String, Any?>(
+            "protocolVersion" to ProtocolVersion.latestVersion,
+            "id" to id.toBuffer(),
+            "balance" to 10,
+            "revision" to 0,
+            "publicKeys" to listOf(
+                mapOf<String, Any?>(
+                    "id" to 0,
+                    "type" to IdentityPublicKey.TYPES.ECDSA_SECP256K1,
+                    "data" to Converters.fromBase64("AuryIuMtRrl/VviQuyLD1l4nmxi9ogPzC9LT7tdpo0di"),
+                ),
+                mapOf(
+                    "id" to 1,
+                    "type" to IdentityPublicKey.TYPES.ECDSA_SECP256K1,
+                    "data" to Converters.fromBase64("A8AK95PYMVX5VQKzOhcVQRCUbc9pyg3RiL7jttEMDU+L"),
+                )
+            )
+        )
+
+        return Identity(rawIdentity)
+    }
+
+    fun getIdentityTopUpTransitionFixture(): IdentityTopUpTransition {
+        val rawStateTransition = hashMapOf<String, Any?>(
+            "protocolVersion" to ProtocolVersion.latestVersion,
+            "type" to StateTransition.Types.IDENTITY_CREATE,
+            "assetLockProof" to getInstantAssetLockProofFixture().toObject(),
+            "identityId" to Entropy.generateRandomIdentifier(),
+        )
+
+        return IdentityTopUpTransition(rawStateTransition)
+    }
+
+    fun getPreorderDocumentFixture(options: Map<String, Any?>): Document {
+        val dataContract = getDpnsContractFixture()
+
+        val label = if (options.containsKey("label")) {
+            options["label"] as String
+        } else {
+            "Preorder"
+        }
+        val normalizedLabel = if (options.containsKey("normalizedLabel")) {
+            options["normalizedLabel"] as String
+        } else {
+            label.toLowerCase()
+        }
+
+        val data = hashMapOf<String, Any?>(
+            "label" to label,
+            "normalizedLabel" to normalizedLabel,
+            "parentDomainHash" to "",
+            "preorderSalt" to Entropy.generate(),
+            "records" to hashMapOf(
+                "dashIdentity" to ownerId
+            )
+        )
+
+        data.putAll(options)
+        return documentFactory.create(dataContract, ownerId, "preorder", data)
+    }
+
+    fun getInstantAssetLockProofFixture(oneTimePrivateKey: ECKey = ECKey()): InstantAssetLockProof {
+        val privateKeyHex = "cSBnVM4xvxarwGQuAfQFwqDg9k5tErHUHzgWsEfD4zdwUasvqRVY"
+        val privateKey = ECKey.fromPrivate(Base58.decode(privateKeyHex))
+        // val fromAddress = Address.fromKey(PARAMS, privateKey)
+
+        // val oneTimePublicKey = oneTimePrivateKey.pubKey
```

Comment:
Admittedly, I don't entirely understand the purpose of this code (is this for tests only?), but should there be private keys hardcoded? Also, commented code - better to either clean up or leave some explanation.

Reply:
Yes, the `Fixtures.kt` file is for tests only.  Some private keys will be hard coded in the tests to sign data and to verify signatures.  These tests are taken from the `js-dpp` project, which is the Dash Platform Protocol repo maintained by the Platform team.

I will search through the current code and clean out commented code.